packages/cli/src/auth-gcp.ts

```diff
@@ -0,0 +1,58 @@
+import path from 'node:path';
+import { GoogleAuth } from 'google-auth-library';
+import {
+  fs,
+  question,
+  echo,
+} from 'zx';
+
+// https://developers.google.com/identity/protocols/oauth2/service-account?hl=pt-br
+
+const readFile = (pathFile: string): Promise<any | null> => new Promise((resolve) => {
+  try {
+    const file = fs.readJSONSync(pathFile);
+    resolve(file as any);
+  } catch (e) {
+    resolve(null);
+  }
+});
+
+export default async (projectId: string, pwd: string) => {
+  await fs.ensureDir(path.join(pwd, '.cloudcommerce'));
+  const pathAccountKeys = path.join(pwd, '.cloudcommerce', 'account-keys.json');
+  const accountKeys = await readFile(pathAccountKeys);
+
+  if (!accountKeys || !accountKeys?.clientId
+    || !accountKeys?.clientSecret || !accountKeys?.refreshToken) {
+    await echo`-- Get the Google administrator account credentials.
+    (i) To obtain them, access (https://shell.cloud.google.com/?fromcloudshell=true&show=terminal)
+
+    (ii) Execute 'gcloud auth application-default login' in cloud shell
+
+    (iii) Then access the created credentials file, to obtain them`;
+  }
+
+  const clientId = accountKeys?.clientId || await question('clientId: ');
+  const clientSecret = accountKeys?.clientSecret || await question('clientSecret: ');
+  const refreshToken = accountKeys?.refreshToken || await question('refreshToken: ');
+
+  const clientOptions = {
+    clientId,
+    clientSecret,
+    refreshToken,
+  };
+
+  if (!accountKeys || !accountKeys?.clientId
+    || !accountKeys?.clientSecret || !accountKeys?.refreshToken) {
+    // update file
+    fs.writeJSONSync(pathAccountKeys, clientOptions);
+  }
```

Comment:
Esse setup é uma vez só por projeto, pra quê salvar esse arquivo?

Reply:
Pelo que entendi são credenciais globais da conta no Google Cloud, não é interessante armazenar, se realmente for o caso deveríamos ter que tratar com mais segurança